bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -120,13 +127,20 @@ def get_bcl2fastq_version_from_run_parameters(runfolder, config):
         :return the version of bcl2fastq to use.
         """
 
-        meta_data = InteropMetadata(runfolder)
-        model = meta_data.model
+        run_info = Bcl2FastqConfig.runinfo_as_dict(runfolder)
+        instrument_name = run_info["RunInfo"]["Run"]["Instrument"]
 
-        current_config = config
-        version = current_config["machine_type"][model]["bcl2fastq_version"]
+        machine_type_mappings = {"M": "MiSeq",
+                                 "D": "HiSeq 2500",
+                                 "SN": "HiSeq 2000",
+                                 "ST": "HiSeq X",
+                                 "A": "NovaSeq",
+                                 "NS": "NextSeq 500",
+                                 "K": "HiSeq 4000"}
```

Comment:
In the future, I think this mapping would be nice to have in a config..

Reply:
Yes. I agree. Right now however I think this will have to do.